### DISCUSSION TITLE
Record and Stream at different resolutions

### DISCUSSION THREAD

**Discussion by cdiddy77** [pedroSG94/RootEncoder#1429] [General]
Our users typically stream video at 720p or lower, but would like to record at 1080p. Is this possible with the current API? What sort of changes would I need to make to enable that?

**Comment by pedroSG94** [comment]
Hello,

Currently it is not possible without modify the library. This is because to do it you need 2 VideoEncoders and this could be a performance problem. It is also harder to maintain because you need handle with different aspect ratio if you use it and other cases.
You have other post related with this case here:
https://github.com/pedroSG94/RootEncoder/issues/1365

You have a link to other issue with steps to do it but it is a bit old. Anyway, the process to implement should be really similar.
You only need know that OffScreenGlThread was replaced to GlStreamInterface and ManagerRender to MainRender

This could be hard to implement but add this feature to the library doesn't feels well. Maybe add it with a cap like use same aspect ratio could be an option

**Comment by cdiddy77** [comment]
Wow thanks for getting back to me so fast!

We have a lot going on so I will probably get to look at this in a couple of weeks.
